File: BACKEND/TM.Application/UseCase/TaskItemUseCases/CreateTaskItemUseCase.cs
using AutoMapper;
using FluentValidation;
using MediatR;
using TM.Application.DTO.TaskItem.Create;
using TM.Domain.Entity;
using TM.Domain.Interface;

namespace TM.Application.UseCase.TaskItemUseCases;

public class CreateTaskItemUseCase : IRequestHandler<CreateTaskRequest, CreateTaskResponse>
{
    private readonly IMapper _mapper;
    private readonly ITaskItemRepository _taskItemRepository;
    private readonly IValidator<CreateTaskRequest> _validator;

    public CreateTaskItemUseCase(IMapper mapper, ITaskItemRepository taskItemRepository, IValidator<CreateTaskRequest> validator)
    {
        _mapper = mapper;
        _taskItemRepository = taskItemRepository;
        _validator = validator;
    }

    public async Task<CreateTaskResponse> Handle(CreateTaskRequest request, CancellationToken cancellationToken)
    {
        var validationResponse = await _validator.ValidateAsync(request, cancellationToken);

        if(!validationResponse.IsValid)
        return new CreateTaskResponse{ Response = false, Message = string.Join(", ", validationResponse.Errors.Select(e => e.ErrorMessage)) };

        var response = await _taskItemRepository.CreateTaskItem(_mapper.Map<TaskItem>(request), cancellationToken);

        return new CreateTaskResponse{ Response = response, Message = response ? "Task created" : "Error creating" };
    }
}

File: FRONTEND/TM.Frontend/src/application/useCases/taskItemUseCases/createTaskItemUseCase.ts
import type { ITaskItemRepository } from "../../../domain/interface/ITaskItemRepository";
import type { CreateTaskInput } from "../../task/createTaskInput";


export const createTaskItemUseCase = async (repository: ITaskItemRepository,createTaskInput:CreateTaskInput) : Promise<void> =>{

    if(!createTaskInput.name || !createTaskInput.description)
        throw new Error("Name and description cant be null")

    const result = await repository.createTaskItem(createTaskInput);

    if(!result)
        throw new Error("Error creating new task item");
}